### PR TITLE
Fix relayer security rules

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-cli/issues/1673

### CODE REVIEW

File: cmd/nodecmd/destroy.go [hunk 1]
```diff
@@ -160,6 +164,12 @@ func destroyNodes(_ *cobra.Command, args []string) error {
 				}
 				ux.Logger.PrintToUser("node %s is already destroyed", nodeConfig.NodeID)
 			}
+			if awmRelayerHost != nil {
+				if err := deleteAWSAWMRelayerSecurityGroupRule(ec2Svc, &nodeConfig, awmRelayerHost); err != nil {
+					ux.Logger.RedXToUser("unable to delete IP address %s from security group %s in region %s due to %s, please delete it manually",
```
Comment (arturrez) [inline cmd/nodecmd/destroy.go:169]: nit: use `%w` for err

File: cmd/nodecmd/wiz.go [hunk 2]
```diff
@@ -798,3 +803,110 @@ func filterHosts(hosts []*models.Host, nodes []string) ([]*models.Host, error) {
...
+				lastRegion = nodeConfig.Region
+			}
+			securityGroupExists, sg, err := ec2Svc.CheckSecurityGroupExists(nodeConfig.SecurityGroup)
+			if err != nil {
+				return err
+			}
+			if !securityGroupExists {
+				return fmt.Errorf("security group %s doesn't exist in region %s", nodeConfig.SecurityGroup, nodeConfig.Region)
+			}
+			inSG := awsAPI.CheckUserIPInSg(&sg, awmRelayerHost.IP, constants.AvalanchegoAPIPort)
```
Comment (arturrez) [inline cmd/nodecmd/wiz.go:859]: nit: we should rename CheckUserIPinSg to CheckIPinSg as it's not User anymore

Comment (arturrez) [inline cmd/nodecmd/wiz.go:859]: nit: use if inSG:=; !inSG instead

File: cmd/nodecmd/wiz.go [hunk 3]
```diff
@@ -798,3 +803,110 @@ func filterHosts(hosts []*models.Host, nodes []string) ([]*models.Host, error) {
...
+	}
+	hasGCPNodes := false
+	lastRegion := ""
+	var ec2Svc *awsAPI.AwsCloud
+	for _, cloudID := range clusterConfig.GetCloudIDs() {
+		nodeConfig, err := app.LoadClusterNodeConfig(cloudID)
+		if err != nil {
+			return err
+		}
+		if nodeConfig.CloudService == "" || nodeConfig.CloudService == constants.AWSCloudService {
```
Comment (arturrez) [inline cmd/nodecmd/wiz.go:844]: nit: use switch here with default: cloud is not supported

**Review by arturrez — APPROVED** [review]

File: cmd/nodecmd/destroy.go (hunk 1 above)
Comment (felipemadero) [reply cmd/nodecmd/destroy.go:169]: I believe only Errorf supports %w

**Pull request merged**